public/app/plugins/datasource/graphite/components/TagsSection.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useCallback, useMemo } from 'react';
+import { Dispatch } from 'redux';
+import { GraphiteSegment } from '../types';
+import { GraphiteTag } from '../graphite_query';
+import { GraphiteQueryEditorState } from '../state/store';
+import { getTagsAsSegmentsSelectables } from '../state/providers';
+import { Button, SegmentAsync, useStyles2 } from '@grafana/ui';
+import { actions } from '../state/actions';
+import { GrafanaTheme2 } from '@grafana/data';
+import { css, cx } from '@emotion/css';
+import { mapSegmentsToSelectables } from './helpers';
+import { TagEditor } from './TagEditor';
+import { debounce } from 'lodash';
+
+type Props = {
+  dispatch: Dispatch;
+  tags: GraphiteTag[];
+  addTagSegments: GraphiteSegment[];
+  state: GraphiteQueryEditorState;
+};
+
+/**
+ * Renders all tags and a button allowing to add more tags.
+ *
+ * Options for tag names are reloaded while user is typing with backend taking care of auto-complete
+ * (auto-complete cannot be implemented in front-end because backend returns only limited number of entries)
+ */
+export function TagsSection({ dispatch, tags, state, addTagSegments }: Props) {
+  const styles = useStyles2(getStyles);
+
+  const newTagsOptions = mapSegmentsToSelectables(addTagSegments || []);
+
+  // Options are reloaded while user is typing with backend taking care of auto-complete (auto-complete cannot be
+  // implemented in front-end because backend returns only limited number of entries)
+  const getTagsAsSegmentsOptions = useCallback(
+    async (inputValue: string) => {
+      return await getTagsAsSegmentsSelectables(state, inputValue);
+    },
```

Comment:
same here, is async/wait needed? 
```suggestion
  const getTagsAsSegmentsOptions = useCallback(
    (inputValue: string) => {
      return getTagsAsSegmentsSelectables(state, inputValue);
    },
```

Reply:
- [ ] remove redundant async/await